### PR TITLE
Added example on filling in ghost cell values.

### LINKED ISSUE #892: Add an IBTK example that explains ghost filling algorithms
I was talking about this with @fuhuifang earlier today - she asked me a question about how we should fill ghost values and I didn't have a great answer. We should add a well-documented IBTK example that explains how to use ghost values, e.g.,
```cpp
using ITC = HierarchyGhostCellInterpolation::InterpolationTransactionComponent;
std::vector<ITC> ghost_cell_components(1);
ghost_cell_components[0] = ITC(u_cc_idx,
                               "LINEAR_REFINE",
                               true,
                               "CONSERVATIVE_COARSEN",
                               "LINEAR",
                               false,
                               {}, // u_bc_coefs
                               nullptr);
HierarchyGhostCellInterpolation ghost_fill_op;
ghost_fill_op.initializeOperatorState(ghost_cell_components, patch_hierarchy);
ghost_fill_op.fillData(/*time*/ 0.0);
```
(from `interpolation_01.cpp`). It would be great if we also showed how to do this in a manual way.

### PR DESCRIPTION
**Pull request by abarret**
Fixes #892. I've added an IBTK example that shows two different ways to fill in ghost cells by 

1) converting from cell average quantities to total cell amounts.
2) Interpolating cell amounts to cell nodes.

### CODE REVIEW

**Comment by abarret** [comment]
@fuhuifang Can you take a look at this and check for clarity?

**Comment by fuhuifang** [comment]
Makes sense to me. Thanks!

File: ibtk/examples/GhostCells/TotalAmountRefineAndCoarsen.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+#include "ibtk/app_namespaces.h"
+#include "ibtk/namespaces.h"
+
+#include "CellVariable.h"
+#include "TotalAmountRefineAndCoarsen.h"
+
+namespace IBTK
+{
+std::string TotalAmountRefine::s_object_name = "AMOUNT_CONSTANT_REFINE";
+static const int REFINE_OP_PRIORITY = 0;
```
Comment (drwells) [inline ibtk/examples/GhostCells/TotalAmountRefineAndCoarsen.cpp]: Could we document what priority means in this context here?

**Comment by abarret** [comment]
@drwells, I've updated the comment. I can merge once the build checks out and you approve the PR.

**Pull request merged**